### PR TITLE
Unlock pyomo version(s)

### PR DESCRIPTION
**Pull request by dgarrett622**
--------
Pull Request Description
--------
##### What issue does this change request address?
Continues #74

##### What are the significant changes in functionality due to this change request?
Continuing discussion from #124 allows upgrading pyomo from 5.7 as a dependency

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/HERON/wiki/Code-Standards) for details.
- [x] 4. Automated Tests should pass.
- [x] 5. If significant functionality is added, there must be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large tes.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added, the the analytic documentation must be updated/added.
- [x] 9. If any test used as a basis for documentation examples have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

**Comment by dgarrett622** [comment]
This PR was ready to go until the ravenframework change. Need to wait for #140 before attempting to merge.

File: src/dispatch/pyomo_dispatch.py [hunk 1]
```diff
@@ -518,11 +517,11 @@ def _create_capacity_constraints(self, m, comp, prod_name, meta):
     r = m.resource_index_map[comp][cap_res] # production index of the governing resource
     caps, mins = self._find_production_limits(m, comp, meta)
     # capacity
-    max_rule = partial(self._capacity_rule, prod_name, r, caps)
+    max_rule = lambda mod, t: self._capacity_rule(prod_name, r, caps, mod, t)
```
Comment (dylanjm) [inline src/dispatch/pyomo_dispatch.py:520]: Do you happen to know which conditional this ends up triggering in the pyomo code? Previously, I had seen an explicit check for a partial function object. I'm just curious what execution path this ends up taking now that it's a lambda. Here is the reference I'm talking about:

https://github.com/Pyomo/pyomo/blob/0edd35481feb2aa9d1f4e46ea231e84462fede42/pyomo/core/base/initializer.py#L112

```python
if type(init) is functools.partial:
        _args = inspect.getfullargspec(init.func)
        if len(_args.args) - len(init.args) == 1 and _args.varargs is None:
            return ScalarCallInitializer(init)
        else:
            return IndexedCallInitializer(init)
```

Comment (dgarrett622) [reply src/dispatch/pyomo_dispatch.py:520]: It looks like it now goes through line 51 where it checks if it is a function or method:

[https://github.com/Pyomo/pyomo/blob/0edd35481feb2aa9d1f4e46ea231e84462fede42/pyomo/core/base/initializer.py#L51](https://github.com/Pyomo/pyomo/blob/0edd35481feb2aa9d1f4e46ea231e84462fede42/pyomo/core/base/initializer.py#L51)

```python
  if inspect.isfunction(init) or inspect.ismethod(init):
      if not allow_generators and inspect.isgeneratorfunction(init):
          raise ValueError("Generator functions are not allowed")
      # Historically pyomo.core.base.misc.apply_indexed_rule
      # accepted rules that took only the parent block (even for
      # indexed components).  We will preserve that functionality
      # here.
      _args = inspect.getfullargspec(init)
      _nargs = len(_args.args)
      if inspect.ismethod(init) and init.__self__ is not None:
          # Ignore 'self' for bound instance methods and 'cls' for
          # @classmethods
          _nargs -= 1
      if _nargs == 1 and _args.varargs is None:
          return ScalarCallInitializer(
              init, constant=not inspect.isgeneratorfunction(init))
      else:
          return IndexedCallInitializer(init)
```

**Comment by PaulTalbot-INL** [comment]
@dgarrett622 This passed tests but is marked with "do not merge"; any opposition to removing that flag?

**Comment by dgarrett622** [comment]
@PaulTalbot-INL no problems with removing the flag and merging. I tried removing the flag earlier, maybe it didn't stick.

**Review by dylanjm — APPROVED** [review]
Changes look good and should not affect functionality or results.

**Pull request merged**